src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -507,36 +520,19 @@ void IdealGraphPrinter::visit_node(Node* n, bool edges) {
     }
 
     const jushort flags = node->flags();
-    if (flags & Node::Flag_is_Copy) {
-      print_prop("is_copy", "true");
-    }
-    if (flags & Node::Flag_rematerialize) {
-      print_prop("rematerialize", "true");
-    }
-    if (flags & Node::Flag_needs_anti_dependence_check) {
-      print_prop("needs_anti_dependence_check", "true");
-    }
-    if (flags & Node::Flag_is_macro) {
-      print_prop("is_macro", "true");
-    }
-    if (flags & Node::Flag_is_Con) {
-      print_prop("is_con", "true");
-    }
-    if (flags & Node::Flag_is_cisc_alternate) {
-      print_prop("is_cisc_alternate", "true");
-    }
-    if (flags & Node::Flag_is_dead_loop_safe) {
-      print_prop("is_dead_loop_safe", "true");
-    }
-    if (flags & Node::Flag_may_be_short_branch) {
-      print_prop("may_be_short_branch", "true");
-    }
-    if (flags & Node::Flag_has_call) {
-      print_prop("has_call", "true");
-    }
-    if (flags & Node::Flag_has_swapped_edges) {
-      print_prop("has_swapped_edges", "true");
-    }
+    const IdealGraphPrintRecord rec[] = {
+        {((flags & Node::Flag_is_Copy) != 0), "is_copy", "true"},
+        {((flags & Node::Flag_rematerialize) != 0), "rematerialize", "true"},
+        {((flags & Node::Flag_needs_anti_dependence_check) != 0), "needs_anti_dependence_check", "true"},
+        {((flags & Node::Flag_is_macro) != 0), "is_macro", "true"},
+        {((flags & Node::Flag_is_Con) != 0), "is_con", "true"},
+        {((flags & Node::Flag_is_cisc_alternate) != 0), "is_cisc_alternate", "true"},
+        {((flags & Node::Flag_is_dead_loop_safe) != 0), "is_dead_loop_safe", "true"},
+        {((flags & Node::Flag_may_be_short_branch) != 0), "may_be_short_branch","true"},
+        {((flags & Node::Flag_has_call) != 0), "has_call", "true"},
+        {((flags & Node::Flag_has_swapped_edges) != 0), "has_swapped_edges", "true"}
+      };
```

Comment:
This code desperately needs a utility to check the corresponding bits in `flags` :-).